### ISSUE TITLE
[java] CommentDefaultAccessModifier - False Positive with JUnit5's ParameterizedTest

### ISSUE THREAD

**Issue by koalalam** [pmd/pmd#4645]
**Affects PMD Version**: 6.55.0, 7.0.0-rc4

**Rule**: [CommentDefaultAccessModifier](https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#commentdefaultaccessmodifier)

**Description:**

Package  defined in ignoredAnnotations is not correct.

Should: org.junit.jupiter.**params**.ParameterizedTest
But: org.junit.jupiter.**api**.ParameterizedTest

**Comment by oowekyala** [comment]
This is fixed in PMD 7: https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#commentdefaultaccessmodifier

**Comment by koalalam** [comment]
Hi @oowekyala ,

To clarify:
1. org.junit.jupiter.**params**.ParameterizedTest is not listed on the doc page
2. Is this the implementation in v7? https://github.com/pmd/pmd/blob/pmd_releases/7.0.0-rc4/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/CommentDefaultAccessModifierRule.java#L57
```java
    private static final PropertyDescriptor<List<String>> IGNORED_ANNOTS =
        JavaPropertyUtil.ignoredAnnotationsDescriptor(
            "com.google.common.annotations.VisibleForTesting",
            "android.support.annotation.VisibleForTesting",
            "co.elastic.clients.util.VisibleForTesting",
            "org.junit.jupiter.api.Test",
            "org.junit.jupiter.api.extension.RegisterExtension",
            "org.junit.jupiter.api.ParameterizedTest",  // "org.junit.jupiter.params.ParameterizedTest" is not there
```

**Comment by adangel** [comment]
You are right, the annotation is wrong, it is in the package `org.junit.jupiter.params` and _not_ in `org.junit.jupiter.api`: https://junit.org/junit5/docs/current/api/org.junit.jupiter.params/org/junit/jupiter/params/ParameterizedTest.html

The rule CommentDefaultAccessModifier seems to be the only one, that uses the wrong package - other rules, that mention this annotation are correct.

When fixing this, also the test case needs to be fixed: https://github.com/pmd/pmd/blob/89c7da82f49e09a31d14d65db7b74fa108519348/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/CommentDefaultAccessModifier.xml#L462

Note: If you don't want to wait for a new PMD release, you can simply configure the rule yourself, see https://docs.pmd-code.org/latest/pmd_userdocs_configuring_rules.html

In this special case, you can use:

```xml
<rule ref="category/java/codestyle.xml/CommentDefaultAccessModifier">
    <properties>
        <property name="ignoredAnnotations" value="android.support.annotation.VisibleForTesting,co.elastic.clients.util.VisibleForTesting,com.google.common.annotations.VisibleForTesting,org.junit.jupiter.api.AfterAll,org.junit.jupiter.api.AfterEach,org.junit.jupiter.api.BeforeAll,org.junit.jupiter.api.BeforeEach,org.junit.jupiter.params.ParameterizedTest,org.junit.jupiter.api.RepeatedTest,org.junit.jupiter.api.Test,org.junit.jupiter.api.TestFactory,org.junit.jupiter.api.TestTemplate,org.junit.jupiter.api.extension.RegisterExtension,org.testng.annotations.AfterClass,org.testng.annotations.AfterGroups,org.testng.annotations.AfterMethod,org.testng.annotations.AfterSuite,org.testng.annotations.AfterTest,org.testng.annotations.BeforeClass,org.testng.annotations.BeforeGroups,org.testng.annotations.BeforeMethod,org.testng.annotations.BeforeSuite,org.testng.annotations.BeforeTest,org.testng.annotations.Test" />
        <property name="regex" value="\/\*\s*(default|package)\s*\*\/" />
        <property name="checkTopLevelTypes" value="false" />
    </properties>
</rule>
```
